File: Model/domaine/Article.php
<?php
    // Définition de la classe Article


    class Article {
        private $id; // Identifiant de l'article
        private $titre; // Titre de l'article
        private $contenu; // Contenu de l'article
        private $categorie; // Catégorie de l'article
        private $dateCreation; // Date de création de l'article
        private $dateModification; // Date de modification de l'article



        // Constructeur de la classe

        public function __construct($id, $titre, $contenu, $categorie, $dateCreation, $dateModification) {
            $this->id = $id;
            $this->titre = $titre;
            $this->contenu = $contenu;
            $this->categorie = $categorie;
            $this->dateCreation = $dateCreation;
            $this->dateModification = $dateModification;
        }

        // Getters

        public function getId() {
            return $this->id;
        }

        public function getTitre() {
            return $this->titre;
        }

        public function getContenu() {
            return $this->contenu;
        }

        public function getCategorie() {
            return $this->categorie;
        }

        public function getDateCreation() {
            return $this->dateCreation;
        }

        public function getDateModification() {
            return $this->dateModification;
        }

        // Setters

        public function setId($id) {
            $this->id = $id;
        }

        public function setTitre($titre) {
            $this->titre = $titre;
        }

        public function setContenu($contenu) {
            $this->contenu = $contenu;
        }

        public function setCategorie($categorie) {
            $this->categorie = $categorie;
        }

        public function setDateCreation($dateCreation) {
            $this->dateCreation = $dateCreation;
        }

        public function setDateModification($dateModification) {
            $this->dateModification = $dateModification;
        }
    }
?>

File: service_web_projet/src/main/java/metier/Article.java
package metier;

import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlRootElement;

import java.util.Date;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class Article {
    private Long id;
    private String titre;
    private  String contenu;
    private Date dateCreation;
    private Date dateModification;
    private Long categorie;

    // Constructeurs, getters et setters
    public Article() {}

    public Article(Long id, String titre, String contenu, Date dateCreation, Date dateModification, Long categorie) {
        this.id = id;
        this.titre = titre;
        this.contenu = contenu;
        this.dateCreation = dateCreation;
        this.dateModification = dateModification;
        this.categorie = categorie;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getContenu() {
        return contenu;
    }

    public void setContenu(String contenu) {
        this.titre = contenu;
    }

    public Date getDateCreation() {
        return dateCreation;
    }

    public void setDateCreation(Date dateCreation) {
        this.dateCreation = dateCreation;
    }

    public Date getDateModification() {
        return dateModification;
    }

    public void setDateModification(Date dateModification) {
        this.dateModification = dateModification;
    }
    public Long getCategorie() {
        return categorie;
    }

    public void setCategorie(Long categorie) {
        this.categorie = categorie;
    }
}